File: app/src/main/java/com/testapp/testapp/view/AdapterView.java
package com.testapp.testapp.view;

import android.content.Context;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.ProgressBar;
import android.widget.Toast;

import com.testapp.testapp.presenter.adapter.CommonRecyclerViewAdapter;

import java.util.List;

/**
 * Created by troll on 07.11.2017.
 */

public class AdapterView<T, VH extends RecyclerView.ViewHolder> implements CustomListView<T> {

    private Context context;
    private ProgressBar loading;
    private RecyclerView recyclerView;
    private CommonRecyclerViewAdapter<T, VH> adapter;

    public AdapterView(Context context, ProgressBar progressBar, CommonRecyclerViewAdapter<T, VH> adapter,
                       RecyclerView recyclerView) {
        this.context = context;
        this.recyclerView = recyclerView;
        this.adapter = adapter;
        loading = progressBar;
    }

    @Override
    public void onSuccessResponse(List<T> items) {
        LinearLayoutManager layoutManager = new LinearLayoutManager(context);
        recyclerView.setLayoutManager(layoutManager);
        recyclerView.setAdapter(adapter);
        adapter.addItems(items);

    }

    @Override
    public void onStartRequest() {
        loading.setVisibility(View.VISIBLE);
    }

    @Override
    public void onFailureRequest() {
        loading.setVisibility(View.INVISIBLE);
        Toast.makeText(context, "Unfortunately, can`t load", Toast.LENGTH_SHORT)
                .show();
    }

    @Override
    public void onEndRequest() {
        loading.setVisibility(View.INVISIBLE);
    }
}

File: app/src/main/java/com/testapp/testapp/kotlin/view/AdapterView.kt
package com.testapp.testapp.kotlin.view

import android.content.Context
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import android.view.View
import android.widget.ProgressBar
import android.widget.Toast
import com.testapp.testapp.kotlin.presenter.adapter.CommonRecyclerViewAdapter

/**
 * Created by dmitriysmishnyi on 29.08.18.
 */
class AdapterView<T, VH : RecyclerView.ViewHolder>(private val context: Context, private val progressBar: ProgressBar?,
                                                   val adapter: CommonRecyclerViewAdapter<T, VH>,
                                                   private val recyclerView: RecyclerView?) : CustomListView<T> {
    override fun onStartRequest() {
        progressBar?.visibility = View.VISIBLE
    }

    override fun onFailureRequest() {
        progressBar?.visibility = View.INVISIBLE
        Toast.makeText(context, "Unfortunately, can`t load", Toast.LENGTH_SHORT).show()
    }

    override fun onEndRequest() {
        progressBar?.visibility = View.INVISIBLE
    }

    override fun onSuccessResponse(data: List<T>) {
        val layoutManager = LinearLayoutManager(context)
        recyclerView?.layoutManager = layoutManager
        recyclerView?.adapter = adapter
        adapter.addItems(data)
    }
}